<div align="center">

#           **6-Weeks Summer Internship**  
## **Machine Learning & Generative AI with Python**

---

# **ASSIGNMENT - 3**

</div>

**1. Create a function to print first 10 natural numbers.**


In [ ]:

def printnum():
    for i in range(1,11):
        print(i)
printnum()


1
2
3
4
5
6
7
8
9
10


**2. Create a function to calculate sum of first N natural numbers.**

In [ ]:
def sum(n):
    sum=0
    for i in range(1,n+1):
        sum+=i
    return sum
num=int(input("Enter the value of n:"))
result=sum(num)
print(result)

Enter the value of n:5
15


**3. Create a function to reverse a number.**

In [ ]:
def reverseNum(n):
  reverse_num=0
  while n!=0:
    digit=n%10
    reverse_num=reverse_num*10+digit
    n//=10
  return reverse_num
n=int(input("Enter the Number:"))
result=reverseNum(n)
print(result)




Enter the Number:5678
8765


**4. Create a Function to count digits in a number.**

In [ ]:
def countDigits(n):
  count=0
  num=str(n)
  for i in num:
    count+=1
  return count
n=int(input("Enter the Number:"))
result=countDigits(n)
print(f"The Number of digits in {n} is {result}.")

Enter the Number:12456
The Number of digits in 12456 is 5.


**5. Create a Function to check Palindrome Numbers.**

In [ ]:
def palindrome(n):
  num=str(n)
  reversenum=0
  while n!=0:
    digit=n%10
    reversenum=reversenum*10+digit
    n//=10
  if str(reversenum)==num:
    return True
  else:
    return False
n=int(input("Enter the Number:"))
result=palindrome(n)
print(result)


Enter the Number:676
True


**6. Create a Function to generate Fibonacci series.**

In [5]:
def fibonacci(n):
    a, b = 0, 1
    for i in range(n):
        print(a, end=" ")
        a, b = b, a + b

fibonacci(10)

0 1 1 2 3 5 8 13 21 34 


**7. Calculator Using Functions that contains the following features:<br>
	-	User selects operation<br>
	-	Program performs calculation<br>
	-	Display result**


In [9]:
def calc(a,b,c):
  result=0
  if c=='+':
    result=a+b
  elif c=='-':
    result=a-b
  elif c=='*':
    result=a*b
  else:
    result=a/b
  return result
a=int(input("Enter the first Number:"))
b=int(input("Enter the second Number:"))
c=input("Enter the operator(+,-,*,/):")
result=calc(a,b,c)
print(f"{a}{c}{b}={result}")

Enter the first Number:7
Enter the second Number:7
Enter the operator(+,-,*,/):*
7*7=49


**8. Create a Text File and store Student Details.**

In [ ]:
def store_student_details():
    name = input("Enter Student Name: ")
    roll_no = input("Enter Roll Number: ")
    course = input("Enter Course: ")

    with open("student_details.txt", "w") as file:
        file.write("Student Details\n")
        file.write("-----------------\n")
        file.write(f"Name: {name}\n")
        file.write(f"Roll No: {roll_no}\n")
        file.write(f"Course: {course}\n")

    print("Student details stored successfully.")

store_student_details()

Enter Student Name: Nidhi
Enter Roll Number: 13401012024
Enter Course: B.Tech
Student details stored successfully.


**9. Read data from student_details.txt File.**

In [ ]:
def readfile():
  with open("student_details.txt","r") as f:
    data=f.read()
    print(data)
readfile()

Student Details
-----------------
Name: Nidhi
Roll No: 13401012024
Course: B.Tech



**10. Handle division by zero using exception handling.**

In [8]:
def division(num1,num2):
  try:
    result = num1 / num2
    print("Result =", result)

  except ZeroDivisionError:
    print("Error: Division by zero is not allowed.")
a=int(input("Enter the first Number:"))
b=int(input("Enter the Second Number:"))
division(a,b)

Enter the first Number:2
Enter the Second Number:0
Error: Division by zero is not allowed.


**11. Create a student Class with Name and Marks.**

In [7]:
class Student:
  def __init__(self,name,marks):
    self.name=name
    self.marks=marks
  def display(self):
        print(f"Name: {self.name}")
        print(f"Marks: {self.marks}")
student1 = Student("Alice", 100)
student1.display()

Name: Alice
Marks: 100


**Name:Nidhi Kumari<br>
Enrollment No:13401012024<br>
College Name:IGDTUW**